### PR TITLE
Improve documents formatting and markup overall

### PR DESCRIPTION
**Pull request by Rami-Sabbagh**
Hello Eng. Muhammad :wave:

I've did some improvements on the format of markdown documents.

Overall I've:
- Used codeblocks wherever possible.
- Enabled `prolog` syntax highlighting on the codeblocks.
- Pretty formatted markdown tables.
- Used headings wherever convinient.
- Used inline-codeblocks wherever possible.

I did also fix the `append/3` example.

I hope you find the modifications helpful!

### CODE REVIEW

File: Lecture3.md [hunk 1]
```diff
@@ -137,6 +149,9 @@ run :-  write('Enter the first number: '), read(FirstNumber),
 
 ?- length([1,2,3, ahmad], X).
 
-?- append([1,2,[a, b], X).
```
Comment (MuhammadSulaiman001) [inline Lecture3.md:140]: Hi @Rami-Sabbagh, there is a missing `]` here, It was meant to be:

`append([1,2],[a, b], X).`

File: Lecture3.md [hunk 2]
```diff
@@ -137,6 +149,9 @@ run :-  write('Enter the first number: '), read(FirstNumber),
...
+?- append([1,2,[a, b]], [X, X2, X3]). %append/2 (merge 2 lists)
+%   [1,2,[a,b],X,X2,X3]
 
+?- append([[1,2,4], [5, 6, 7]]). %append/1 (list of lists)
```
Comment (MuhammadSulaiman001) [inline Lecture3.md]: @Rami-Sabbagh, this append/1 won't work in command line, and I think there is no append/1 out there. I suggest you to replace append/1 and append/2 with this append/3 example `append([1,2],[a, b], X).`

**Review by MuhammadSulaiman001 — CHANGES_REQUESTED** [review]

**Comment by MuhammadSulaiman001** [comment]
Hi @Rami-Sabbagh, thank you very much for this work.. I will merge your changes once we resolve the `append` conversation :thumbsup:

File: Lecture3.md (hunk 2 above)
Comment (Rami-Sabbagh) [reply Lecture3.md]: Oh right my mistake, `append/1` is not for lists.

**Comment by Rami-Sabbagh** [comment]
Hey @MuhammadSulaiman001 I've pushed the requested changes. Check them when you get the chance :)

**Comment by MuhammadSulaiman001** [comment]
DONE

**Pull request merged**